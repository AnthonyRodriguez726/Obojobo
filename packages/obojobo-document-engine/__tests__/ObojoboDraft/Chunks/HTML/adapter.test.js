import HtmlAdapter from '../../../../ObojoboDraft/Chunks/HTML/adapter'

describe('HTML adapter', () => {
	test('can be constructed WITHOUT attributes', () => {
		let model = { modelState: {} }
		HtmlAdapter.construct(model)
		expect(model).toMatchSnapshot()
	})

	test('construct builds with attributes', () => {
		let model = { modelState: {} }
		let attrs = { content: { html: 'html', align: 'center' } }
		HtmlAdapter.construct(model, attrs)
		expect(model).toMatchSnapshot()
	})

	test('clone creates a copy', () => {
		let a = { modelState: {} }
		let b = { modelState: {} }

		HtmlAdapter.construct(a)
		HtmlAdapter.clone(a, b)

		expect(a).not.toBe(b)
		expect(a.modelState).toEqual(b.modelState)
	})

	test('toJSON builds a JSON representation', () => {
		let model = {
			modelState: {
				html: 'html',
				align: 'center'
			}
		}
		let json = { content: {} }

		HtmlAdapter.toJSON(model, json)
		expect(json).toMatchSnapshot()
	})

	test('toText creates a text representation', () => {
		let model = { modelState: { html: 'expected text to be returned' } }
		expect(HtmlAdapter.toText(model)).toMatch('expected text to be returned')
	})
})
