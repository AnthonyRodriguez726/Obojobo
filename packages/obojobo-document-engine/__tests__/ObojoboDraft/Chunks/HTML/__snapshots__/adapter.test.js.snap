// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML adapter can be constructed WITH attributes 1`] = `
Object {
  "modelState": Object {
    "align": "center",
    "html": "html",
  },
}
`;

exports[`HTML adapter can be constructed WITHOUT attributes 1`] = `
Object {
  "modelState": Object {
    "align": "left",
    "html": null,
  },
}
`;

exports[`HTML adapter can be converted to JSON 1`] = `
Object {
  "content": Object {
    "align": "center",
    "html": "html",
  },
}
`;
